src/endpoints/ECatCourse/index.ts

```diff
@@ -164,6 +164,52 @@ class ECatCourse extends EndpointCategory {
     });
   }
 
+  /**
+   * Update whether the course is published or not
+   * @author Gabe Abrams
+   * @method updatePublishState
+   * @memberof api.course
+   * @instance
+   * @async
+   * @param {object} [opts] object containing all arguments
+   * @param {number} [opts.courseId=default course id] Canvas course Id to
+   *   modify
+   * @param {boolean} [opts.isPublished] if true, publish the course. Otherwise,
+   *   unpublish the course
+   */
+  public async updatePublishState(
+    opts: {
+      courseId?: number,
+      isPublished?: boolean,
+    } = {},
+    config?: APIConfig,
+  ): Promise<CanvasCourse> {
+    const course = await this.visitEndpoint({
+      config,
+      action: 'update the published state of a specific course',
+      path: `${API_PREFIX}/courses/${opts.courseId ?? this.defaultCourseId}`,
+      method: 'PUT',
+      params: {
+        'course[event]': (
+          opts.isPublished
+            ? 'offer'
+            : 'claim'
```

Comment:
"claim" is a weird word for unpublishing a course. Is that a Canvas thing?

Reply:
Just an observation that using defaultCouseId seems like a dangerous thing for publishing and unpublishing a course.

Reply:
this is the default behavior for all endpoints. Unpublishing a course isn't a dangerous activity because Canvas doesn't allow unpublishing courses that are active.

Reply:
"claim" is unfortunately a Canvas term and I also don't like it